test/python/pulse/test_schedule.py

```diff
@@ -264,6 +270,21 @@ def test_keep_original_schedule_after_attached_to_another_schedule(self):
         # sched must keep 3 instructions (must not update to 5 instructions)
         self.assertEqual(3, len(list(sched.instructions)))
 
+    @patch('qiskit.util.is_main_process', return_value=True)
+    def test_auto_naming(self, is_main_process_mock):
+        """Test that a schedule gets a default name, incremented per instance"""
+
+        del is_main_process_mock
+
+        sched0 = Schedule()
+        self.assertEqual('sched1', sched0.name)
```

Comment:
I can't figure out why the instance counter starts at `1` instead of `0`. This doesn't happen when testing with the REPL, only in these tests. I'm not too familiar with python in general--is there some nuance to the unit testing to the unit testing I'm missing?

Reply:
I believe the test suite does run in parallel (stestr). I worry that this test will not be robust because of this if something on the import path changes it could break the hardcoded name. Perhaps you could create multiple schedules and extract the names and indices and verify that these indices are incrementing?